### PR TITLE
qt5base: add package

### PR DESCRIPTION
**Pull request by yangfl**
Maintainer: me / @mirko 
Compile tested: 19.07.7 ramips

Description:

This package is mainly stolen from the same package from video feed, but
without any GUI functionalities, to enable building for other packages
that depend on Qt core library.

### CODE REVIEW

File: libs/qt5base/src/mkspecs/linux-openwrt-g++/qplatformdefs.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#include "../linux-g++/qplatformdefs.h"
+
+#undef QT_SOCKLEN_T
+
+#if defined(__GLIBC__) && (__GLIBC__ < 2)
```
Comment (neheb) [inline libs/qt5base/src/mkspecs/linux-openwrt-g++/qplatformdefs.h:5]: This looks like a hack for uClibc, which is not in the tree anymore.

**Comment by mirko** [comment]
Thanks for submitting!
For me qt5base already compiles and works fine for headless systems without when only selecting the non-GUI related packages - compiled/linked against musl as well as glibc.
Please elaborate on the issue you're trying to fix here.

**Comment by yangfl** [comment]
This patch is and is only meant to be a dependency for any further package (qbittorrent-nox) that want to go into packages feed, as mentioned in https://github.com/openwrt/packages/pull/15977#issuecomment-874976957 , thus we must have qtcore in packages feed as well.

**Comment by yangfl** [comment]
Failling from #16116

**Comment by ryzhovau** [comment]
> This patch is and is only meant to be a dependency for any further package (qbittorrent-nox) that want to go into packages feed

QBittorrent proven as working in @Entware, but it's still too heavy for MIPS devices. 
https://github.com/openwrt/packages/issues/10125

**Comment by yangfl** [comment]
But arm...

File: libs/qt5base/Makefile [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+	--egl=no \
+	--xcb-xlib=no \
+	--qpa=$(if $(CONFIG_PACKAGE_qt5base-plugin-platforms-linuxfb),linuxfb,no) \
+	--directfb=no \
+	--eglfs=no \
+	--gbm=no \
+	--linuxfb=$(if $(CONFIG_PACKAGE_qt5base-plugin-platforms-linuxfb),yes,no) \
+	--xcb=no \
+	\
+	--libudev=no \
```
Comment (Lankaster) [inline libs/qt5base/Makefile:199]: What about qt5-serial ?I think qt5-serial need libudev, or not ?

File: libs/qt5base/Makefile [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+	--prefix=$(STAGING_DIR_HOST) \
+	--hostprefix=$(STAGING_DIR_HOST) \
+	--hostdatadir=$(STAGING_DIR_HOST)/share \
+	--datadir=$(STAGING_DIR_HOST)/share \
+	--archdatadir=$(STAGING_DIR_HOST)/lib \
+	--compile-examples=no \
+	--make=libs \
+	--nomake=tools \
+	--nomake=examples \
+	--gui=yes \
```
Comment (Lankaster) [inline libs/qt5base/Makefile:269]: Did you really need GUI ?

**Review by Lankaster — COMMENTED** [review]
Wrong name & some question

**Comment by Lankaster** [comment]
Another important thing: 
```
  Target compiler supports:
    SSE .................................. SSE2 SSE3 SSSE3 SSE4.1 SSE4.2
    AVX .................................. AVX AVX2
    AVX512 ............................... F ER CD PF DQ BW VL IFMA VBMI
    Other x86 ............................ AES F16C RDRAND SHA
    Intrinsics without -mXXX option ...... yes
```
... also double-conversion ([source](https://github.com/google/double-conversion)) library support should be deactivated

**Comment by mirko** [comment]
I'm still not sure what this pull request is being about.
We have a maintained qt5base package as part of the video feed (among others): https://github.com/openwrt/video/tree/master/frameworks/qt5base
If you want to apply changes to that, please open up pull-requests over there and don't copy and adjust Makfiles across feeds within the same pull request.
If this is about moving qt5base from the video feed to the packages feed, I'm happy to talk and discuss about that as well, but please keep such intentions separated. This PR is the wrong place for such discussion.

**Comment by yangfl** [comment]
I started https://github.com/openwrt/packages/pull/15977 based on codes from external repos, without even notice the existance of video feed at the first place. I'm happy to have someone else maintaining this library, but that being in video feed means I can't submit any packages into packages feed which depend on it, nor I see any efforts on moving qt5base into packages. So it's totally up to you which work flow is better on adding (or moving or whatsoever) qt5base, and I'm willing to have the discussion in the right place (as long as I'm correctly informed), but having no progress for several months is really depressing.

**Comment by mirko** [comment]
What do you mean by "no progress"?
In your very initial comment of this PR, you're literally writing, quote: "This package is mainly stolen from the same package from video feed" - while the third comment is already mine, asking for elaboration.

**Comment by Lankaster** [comment]
@mirko Im sorry, I didn't know that the QT was already in openwrt feeds.
This pull request contains serialbus and serialport plugins and i'd like to have it.

**Comment by mirko** [comment]
> @mirko Im sorry, I didn't know that the QT was already in openwrt feeds.
> This pull request contains serialbus and serialport plugins and i'd like to have it.

If you could make it (easily) work with what's already upstream (https://github.com/openwrt/video/tree/master/frameworks) that would make things easy and fast.

If your changes depend on packages as part of the packages feed, then we obviously first would have to finish up on the move from the video/ into the packages/ feed. Here again, feel free to start a discussion on the openwrt-devel ML. Mentioning your use cases will certainly help.

**Comment by mirko** [comment]
If you feel like getting in touch with directly about Qt5 and OpenWrt and why we currently have the video feed: feel free to query me on the usual IRC networks, me always just being "mirko".

**Comment by Lankaster** [comment]
@mirko I believe that working with upstream (https://github.com/openwrt/video/tree/master/frameworks) make more sense for me.

**Comment by BKPepe** [comment]
Closing as qt5base is in the [video feed](https://github.com/openwrt/video). For more details take a look at this comment: https://github.com/openwrt/packages/pull/16114#issuecomment-922516659

**Pull request closed without merging**